### PR TITLE
Adding seed tests #335

### PR DESCRIPTION
**Pull request by duckyou**
1. Grouped all tests using classes.
2. Added tests for seeded data providers.

### CODE REVIEW

File: tests/test_data/test_address.py [hunk 1]
```diff
@@ -10,151 +10,220 @@
...
+            assert a1.street_suffix() == a2.street_suffix()
+
+    def test_address(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
+            assert a1.address() == a2.address()
+
+    def test_state(self, _addresses):
+        a1, a2 = _addresses
+        for _ in range(self.TIMES):
```
Comment (lk-geimfari) [inline tests/test_data/test_address.py]: Are there no other much elegant way to repeat test `n` time? Something like:
```
@pytest.mark.repeat(self.TIMES)
def test_state(self, _addresses):
    a1, a2 = _addresses
    assert a1.state() == a2.state()
    assert a1.state(abbr=True) == a2.state(abbr=True)
```

Comment (sobolevn) [reply tests/test_data/test_address.py]: Why do even need to repeat a test?

File: mimesis/providers/base.py [hunk 2]
```diff
@@ -9,7 +9,7 @@ class BaseProvider(object):
     """This is a base class for all providers.
     """
 
-    def __init__(self, seed: Optional[int] = None) -> None:
+    def __init__(self, seed: Union[int, str, bytes, bytearray] = None) -> None:
```
[the remark this answers is not in the document]
Comment (sobolevn) [reply mimesis/providers/base.py]: It is better to use a custom type here:

```python
Seed = Union[int, str, bytes, bytearray]  # or whatever it is
```

File: tests/test_data/test_address.py (hunk 1 above)
Comment (lk-geimfari) [reply tests/test_data/test_address.py]: I think to ensure that result is same for each iteration.

File: mimesis/providers/internet.py [hunk 3]
```diff
@@ -119,9 +119,8 @@ def emoji(self) -> str:
         """
         return self.random.choice(EMOJI)
 
-    @staticmethod
-    def image_placeholder(width: Size = 400,
-                          height: Size = 300) -> str:
+    # TODO: https://en.wikipedia.org/wiki/List_of_common_resolutions
```
Comment (sobolevn) [inline mimesis/providers/internet.py]: You can specify `todo`s in code: https://thomas-cokelaer.info/tutorials/sphinx/docstring_python.html

Comment (lk-geimfari) [reply mimesis/providers/internet.py]:
```rst
.. todo:: https://en.wikipedia.org/wiki/List_of_common_resolutions
```

File: tests/test_data/test_address.py (hunk 1 above)
Comment (duckyou) [reply tests/test_data/test_address.py]: @lk-geimfari If I understand `pytest` correctly, when we writing tests with decorator, providers initialization at runtime will be at every iteration.

Comment (lk-geimfari) [reply tests/test_data/test_address.py]: @duckyou You're right. My idea was not really correct or rather really incorrect.

**Comment by lk-geimfari** [comment]
@duckyou Can we merge it? Is it done?

**Comment by duckyou** [comment]
@lk-geimfari It seems like yes 😄 
@lk-geimfari and @sobolevn Also big thanks for your reviews, guys!  :wink:

**Comment by lk-geimfari** [comment]
@duckyou Thank you for great work!

**Pull request merged**